src/nools/task-emitter.js

```diff
@@ -6,7 +6,7 @@ function taskEmitter(taskDefinitions, c, Utils, Task, emit) {
   var taskDefinition, r;
   for (var idx1 = 0; idx1 < taskDefinitions.length; ++idx1) {
     taskDefinition = taskDefinitions[idx1];
-    prepareDefinition(taskDefinition);
+    prepareDefinition(taskDefinition, defaultResolvedIf, Utils);
```

Comment:
We need to expose `defaultResolvedIf` in the task, right?

I think it comes more natural if defaults for tasks has a similar shape as a task definition and then we can concatenate both objects before passing it to `prepareDefinition()`, what do you think?

So, for example, let's say we have a file containing the definitions of defaults that we can export:
```
// File defining defaults

// Exporting obj similar as task
const taskDefaults = { 
   defaultResolvedIf: function(.....) {
      // do stuff;
   },
   
   // Here we can safely deprecate with backward compatibility or add more defaults in the future since objects can escalate easier.

}

exports {
   taskDefaults
}
```

Then we import `task defaults` and concatenate both objects:
```
// Task emitter file
 taskDefinition = { ...taskDefinitions[idx1], ...taskDefaults }; // <-- test if this works fine.
 prepareDefinition(taskDefinition);  // <-- it didn't changed
```

Unit testing will be more isolated because we just import the file for task defaults and test what it does (single responsibility), therefore the complexity of testing the task-emitter is reduced.

Reply:
Thanks for this great suggestion. It makes things more manageable. So the users will have to use `this.definition.defaultResolvedIf` instead of `this.defaultResolvedIf`? Should not be a problem if we document it nicely. Thanks.

Reply:
`this.definition.defaultResolvedIf`
I think it explains better where fn and props are coming from and keeps code simple and easy to maintain, but usability is important too. 
I'll leave it up to you on what works better for the end user :) Thank you for following up!